docs/user-guide/src/capm3/remediaton.md

```diff
@@ -0,0 +1,143 @@
+# Remediation Controller and MachineHealthCheck
+
+The ```Cluster API``` includes [MachineHealthCheck(MHC)](https://cluster-api.sigs.k8s.io/tasks/healthcheck.html) resource that implements automated health checking feature.  It is possible to plug-in Metal3 specific remediation strategies to remediate an unhealthy nodes while depending on ```Cluster API``` MHC resource to determine those nodes as unhealthy by using ```CAPM3 Remediation Controller```.
+
+## Remediation Controller
+
+CAPM3 ```Remediation Controller(RC)``` reconciles ```Metal3Remediation``` objects created by CAPI MachineHealthCheck. The RC locates a Machine with the same name as the Metal3Remediation CR and uses BMO and CAPM3 APIs to remediate associated unhealthy baremetal nodes. Remediation controller supports ```reboot strategy``` specified in Metal3Remediation CRD and uses the same object to store state of the current remediation cycle.
+
+## CAPI MachineHealthCheck
+
+A MachineHealthCheck is a Cluster API resource, which allows users to define conditions under which Machines within a Cluster should be considered unhealthy. Users can also specify a timeout for each of the conditions that they define to check on the Machine’s Node. If any of these conditions are met for the duration of the timeout, the Machine will be remediated. CAPM3 will use MHC to create remediation requests based on ```Metal3RemediationTemplate``` and ```Metal3Remediation``` CRDs to plug-in remediation solution.
+
+### Remediation workflow
+
+* CAPM3 ```Remediation Controller(RC)``` detects the presence of ```Metal3Remediation``` CR.
+* Remediation strategy can be defined under ```.spec.strategy.type``` in Metal3Remediation, RC uses BMO APIs to get hosts back into a healthy or manageable state.
+* RC uses ```.status.phase``` to save the states of the remediation. Available states are ```running```, ```waiting```, ```deleting machine```.
+* After RC have finished its remediation, it will wait for the Metal3Remediation CR to be removed. MHC will noticed the Node becomes healthy and deletes the instantiated MachineRemediation CR using CAPI MachineHealthCheck controller.
```

Comment:
```suggestion
* After RC has finished its remediation, it will wait for the Metal3Remediation CR to be removed. MHC will notice the Node to become healthy and delete the instantiated MachineRemediation CR using CAPI MachineHealthCheck controller.
```

Reply:
This part is removed